fix: pxe_oracle_interface test flake

benesjan opened the pull request:
In this PR I (hopefully) fix [these flakes](http://ci.aztec-labs.com/list/history_ede44f6f60adaef4_next).

The issue was most likely caused by us producing an incorrect random TxEffect in a test. Incorrect because it had 0 nullifiers and a tx effect needs to always have at least 1 (tx hash nullifier).

Then [here](https://github.com/AztecProtocol/aztec-packages/blob/5b4f20de4dc0e85d71b226222e78d39be115c44b/yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.ts#L573) we got undefined and then later one when the object was getting serialized we got the error.

All credit goes to AI 😆 🤖

benesjan commented:
* **#18682** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18682?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#18678** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18678?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18678?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan commented on yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.ts:
```diff
@@ -756,7 +756,7 @@ export class PXEOracleInterface implements ExecutionDataProvider {
           );
         } else {
-          null;
+          return null;
```
This was an unrelated bug.

mverzilli replied:
Someone didn't context switch from Noir/Rust :P

benesjan commented on yarn-project/pxe/src/contract_function_simulator/noir-structs/log_retrieval_response.test.ts:
```diff
@@ -116,7 +116,7 @@ describe('LogRetrievalResponse', () => {
 
   it('serialization of some matches snapshots and output of Noir serialization', () => {
-    const serialized = LogRetrievalResponse.toSerializedOption();
+    const serialized = LogRetrievalResponse.toSerializedOption(null);
```
Made serialization to an empty log response more explicit by enforcing `null` type on the input.

benesjan commented on yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.test.ts:
```diff
@@ -184,7 +184,7 @@ describe('PXEOracleInterface', () => {
       aztecNode.getLogsByTags.mockReset();
       aztecNode.getTxEffect.mockResolvedValue({
-        ...randomInBlock(await TxEffect.random()),
+        ...randomInBlock(await TxEffect.random({ numNullifiers: 1 })),
```
This is the fix.

benesjan commented on yarn-project/stdlib/src/tx/tx_effect.ts:
```diff
@@ -213,6 +213,9 @@ export class TxEffect {
     maxEffects?: number;
   } = {}): Promise<TxEffect> {
+    if (numNullifiers < 1) {
+      throw new Error(`A tx effect must have at least one nullifier. Only ${numNullifiers} got requested.`);
```
Added this check to prevent these bugs in other places.

benesjan committed:
fix: pxe_oracle_interface test flake

In this PR I (hopefully) fix [these flakes](http://ci.aztec-labs.com/list/history_ede44f6f60adaef4_next).

The issue was most likely caused by us producing an incorrect random TxEffect in a test. Incorrect because it had 0 nullifiers and a tx effect needs to always have at least 1 (tx hash nullifier).

Then [here](https://github.com/AztecProtocol/aztec-packages/blob/5b4f20de4dc0e85d71b226222e78d39be115c44b/yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.ts#L573) we got undefined and then later one when the object was getting serialized we got the error.

All credit goes to AI 😆 🤖
